Compute melt volume (#1972)
chaos-polymtl/lethe: doc/source/parameters/cfd/post_processing.rst

		0.0000          0.0000               0.0000               0.0000            1000.0000 
		1.0000         -0.9732               0.0000               1.4856               0.9732 

* ``calculate liquid fraction``: calculates the liquid fraction in simulations with phase change. The liquid fraction corresponds to the volume (area) integral of the liquid divided by the volume (area) of the domain.

  * ``liquid fraction name``: name of the output file containing the time evolution of the liquid fraction.

* ``calculate melt volume``: calculates the melt volume in simulations with phase change. The melt volume corresponds to the volume (area) integral of the liquid phase of the fluid with phase change. The melt region (:math:`\Omega_\mathrm{melt}`) is described by temperatures above the fluid's liquidus temperature (:math:`T_\mathrm{liquidus}`):

  .. math::
    \Omega_\mathrm{0,melt} = \{\vec{x} \in \Omega_0 | T > T_\mathrm{liquidus} \}

>   here, fluid 0 is a fluid a with phase change and :math:`\Omega_0` corresponds to the subdomain occupied by fluid 0.

Reviewer: So if I understand correctly, one is the volume ABOVE the liquidus and the other one is the volume taking into account the mushy zone. Am I correct?

Author: I've changed the description. I hope it is more clear now with the user-set temperature limit.

  .. warning::

>     This is compatible with CLS two-phase flows. A geometrical volume is computed delimited by the :math:`0.5` phase indicator iso-curve. However, only one of the two fluids can have a ``phase_change`` specific heat model.

Reviewer: Is it always fluid 0 then? or it can be fluid 1? or this is determined at run time and it is the one that has a phase change? I would add more precisions here.
Does this only work with the CLS model,  or it also works in single phase? I would think it works also on single phase?

Author: It works for both CLS and single-fluid flows. Now the fluid is chosen with `monitored fluid with phase change`, before it was checked internally with the specific heat model. :)

		0.0000          0.0000               0.0000               0.0000            1000.0000 
		1.0000         -0.9732               0.0000               1.4856               0.9732 

> * ``calculate liquid fraction``: calculates the liquid fraction in simulations with phase change. The liquid fraction corresponds to the volume (area) integral of the liquid divided by the volume (area) of the domain.

Reviewer: This one is normalized by the volume and the new one is not. Maybe it would be better to refactor this one so that it is not normalized also, to ensure that at least both definitions are consistent with one another.

  * ``liquid fraction name``: name of the output file containing the time evolution of the liquid fraction.

* ``calculate melt volume``: computes the melt volume (surface, in 2D) and contour area (length, in 2D) in simulations with phase change. The melt volume corresponds to the volume integral of the melted phase of the fluid with phase change. The melt region (:math:`\Omega_\mathrm{i,melt}` with :math:`i \in \{0,1\}` depending on the fluid) is characterized by temperatures above the fluid's melting temperature (:math:`T_\mathrm{melt}`):

> 

Reviewer: Can you explain clearly the difference between the melt volume and the liquid fraction other than the fact that they are normalized?
one is geometric the other one is calculated by integrating the mushy zone also? Try to make it explicit what is the difference.
Maybe then we should deprecate the liquid fraction parameter and calculation if this new feature is more "adequate" than the previous one. that's also something I would consider. In that case, we should adapt the melting cavity example to use this new feature instead.

I am not saying this needs to be done, but the parameter description should clearly explain the difference between the two calculation methods and when one wants to use one or the other.

		0.0000          0.0000               0.0000               0.0000            1000.0000 
		1.0000         -0.9732               0.0000               1.4856               0.9732 

* ``monitored fluid with phase change``:  fluid with phase change properties, either ``fluid 0`` or ``fluid 1`` (for CLS multiphase flows). The current implementation does not track two melt volumes simultaneously.

> * ``calculate algebraic melt volume``: calculates the algebraic melt volume in simulations with phase change. The algebraic melt volume (:math:`V_\mathrm{i,melt,alge}` with :math:`i \in \{0,1\}` depending on the fluid) corresponds to the volume (area, in 2D) integral of the liquid of the specified fluid (with ``monitored fluid with phase change``, by default: ``fluid 0``) in the domain.

Reviewer 2: ```suggestion
* ``calculate algebraic melt volume``: calculates the algebraic melt volume in simulations with phase change. The algebraic melt volume (:math:`V_\mathrm{i,melt,alge}` with :math:`i \in \{0,1\}` depending on the monitored fluid) corresponds to the volume (surface, in 2D) integral of the liquid fraction in the specified ``monitored fluid with phase change`` domain.
```

> * ``calculate algebraic melt volume``: calculates the algebraic melt volume in simulations with phase change. The algebraic melt volume (:math:`V_\mathrm{i,melt,alge}` with :math:`i \in \{0,1\}` depending on the fluid) corresponds to the volume (area, in 2D) integral of the liquid of the specified fluid (with ``monitored fluid with phase change``, by default: ``fluid 0``) in the domain.

Reviewer 2: Can you add a warning that phase change as to be enable?

  * For single-phase flows:

    .. math::
      V_\mathrm{0,melt,alge} = \int_{\Omega_0} \alpha_\mathrm{0,liquid} \, \mathrm{d}\Omega

    where :math:`\alpha_\mathrm{0,liquid} = \frac{T-T_\mathrm{0,solidus}}{T_\mathrm{0,liquidus} - T_\mathrm{0,solidus}}` is the liquid fraction of ``fluid 0`` with :math:`T_\mathrm{0,solidus}` and :math:`T_\mathrm{0,liquidus}` respectively the solidus and liquidus temperatures of ``fluid 0``, and :math:`T` is the temperature.

  * For CLS flows:

    .. warning::

      This is compatible with CLS two-phase flows. However, only one of the two fluids can be a fluid identified with phase change using the parameter ``monitored fluid with phase change``.

    * If ``fluid 0`` is monitored:

      .. math::
        V_\mathrm{0,melt,alge} = \int_{\Omega_0} (1-\phi) \, \alpha_\mathrm{0,liquid} \, \mathrm{d}\Omega

    where :math:`\phi` is the phase indicator.

    * If ``fluid 1`` is monitored:

      .. math::
        V_\mathrm{1,melt,alge} = \int_{\Omega_1} \phi \, \alpha_\mathrm{1,liquid} \, \mathrm{d}\Omega

> 

Reviewer 2: I think we could simplify this:

```suggestion

    .. math::
      V_\mathrm{i,melt,alge} = \int_{\Omega_i} \alpha_\mathrm{i,liquid} \, \mathrm{d}\Omega

    where :math:`\alpha_\mathrm{i,liquid} = \frac{T-T_\mathrm{i,solidus}}{T_\mathrm{i,liquidus} - T_\mathrm{i,solidus}}` is the liquid fraction in the fluid :math:`i`,  with :math:`T_\mathrm{i,solidus}` and :math:`T_\mathrm{i,liquidus}` respectively the solidus and liquidus temperatures of fluid :math:`i`, and :math:`T` is the temperature.

    .. warning::

      This is compatible with CLS two-phase flows. However, only one of the two fluids can be a fluid identified with phase change using the parameter ``monitored fluid with phase change``.

```

  * For single-phase flows:

    .. math::
      \Omega_\mathrm{0,melt,geo} = \{\vec{x} \in \Omega | T > T_\mathrm{melt} \}

    here, ``fluid 0`` is considered as a fluid a with phase change, hence the index :math:`0`.

    .. math::
          V_\mathrm{0,melt,geo} = \int_{\Omega_\mathrm{0,melt,geo}} \mathrm{d}\Omega

  * For CLS flows, a geometrical volume is computed delimited by the :math:`0.5` phase indicator isocontour:

  .. warning::

    This is compatible with CLS two-phase flows. However, only one of the two fluids can be a fluid identified with phase change using the parameter ``monitored fluid with phase change``.

  * If ``fluid 0`` is monitored:

    .. math::
      \Omega_\mathrm{0,melt,geo} = \{\vec{x} \in \Omega| T > T_\mathrm{melt}, \, \phi < 0.5 \}

    where :math:`\phi` is the phase indicator.

    .. math::
        V_\mathrm{0,melt,geo} = \int_{\Omega_\mathrm{0,melt,geo}} \mathrm{d}\Omega

  * If ``fluid 1`` is monitored:

    .. math::
      \Omega_\mathrm{1,melt,geo} = \{\vec{x} \in \Omega | T > T_\mathrm{melt}, \, \phi > 0.5 \}

    .. math::

>         V_\mathrm{1,melt,geo} = \int_{\Omega_\mathrm{1,melt,geo}} \mathrm{d}\Omega

Reviewer 2: Same simplification here :)

  * ``geometric melt volume name``: name of the output file containing the time evolution of the geometric melt volume.

>   * ``melting temperature``: temperature in Kelvin above which the fluid is considered melted.

Reviewer 2: ```suggestion
  * ``melting temperature``: temperature above which the fluid is considered melted.
```
